### PR TITLE
Add fluent API and Initial migration

### PR DESCRIPTION
**Pull request by VitalikMatvieiev**

### CODE REVIEW

File: backend/Recodo.API/Startup.cs [hunk 1]
```diff
@@ -52,6 +52,17 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+            InitializeDatabase(app);
+        }
+
+        private static void InitializeDatabase(IApplicationBuilder app)
+        {
+            using (var scope = app.ApplicationServices.GetService<IServiceScopeFactory>().CreateScope())
+            {
+                using var context = scope.ServiceProvider.GetRequiredService<RecodoDbContext>();
+                context.Database.Migrate();
+            };
```
Comment (Jyuart) [inline backend/Recodo.API/Startup.cs]:
```suggestion
            using var scope = app.ApplicationServices.GetRequiredService<IServiceScopeFactory>().CreateScope();
            using var context = scope.ServiceProvider.GetRequiredService<RecodoDbContext>();
            context.Database.Migrate();
```
You can use 'using' declaration in both cases. The same for GetRequiredService

File: backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Recodo.DAL.Entities.Configuration
+{
+    public class CommentReactionConfiguration : IEntityTypeConfiguration<CommentReaction>
```
Comment (Jyuart) [inline backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs]: I think you can configure these as owned entities.
https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities
Cause comment reactions are strictly dependents on the comments

Just put a collection of CommentReactions in the Comment entity and then configure in the CommentConfiguration with builder.OwnsMany()

File: backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        public void Configure(EntityTypeBuilder<CommentReaction> builder)
+        {
+            builder.HasKey(p => p.Id);
+            builder.HasOne(p=>p.User).WithMany().HasForeignKey(p=>p.UserId);
```
Comment (Jyuart) [inline backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs]:
```suggestion
            builder.HasOne(p => p.User).WithMany().HasForeignKey(p=>p.UserId);
```
Cleanliness. Here and everywhere else

File: backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs (hunk 2 above)
Comment (Jyuart) [reply backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs]: We can even make Comment as an owned entity of the video, but let's not make it like this for now.

File: backend/Recodo.DAL/Entities/Configuration/NotificationConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Recodo.DAL.Entities.Configuration
+{
+    public class NotificationConfiguration : IEntityTypeConfiguration<Notification>
+    {
+        public void Configure(EntityTypeBuilder<Notification> builder)
```
Comment (Jyuart) [inline backend/Recodo.DAL/Entities/Configuration/NotificationConfiguration.cs:13]: Type, CreatedAt and IsRead?

File: backend/Recodo.DAL/Entities/Configuration/NotificationConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            builder.HasKey(p => p.Id);
+            builder.Property(p => p.Massage).IsRequired();
```
Comment (Jyuart) [inline backend/Recodo.DAL/Entities/Configuration/NotificationConfiguration.cs]:
```suggestion
            builder.Property(p => p.Message).IsRequired();
```

File: backend/Recodo.DAL/Entities/Configuration/TeamInvitationConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Recodo.DAL.Entities.Configuration
+{
+    public class TeamInvitationConfiguration : IEntityTypeConfiguration<TeamInvitation>
```
Comment (Jyuart) [inline backend/Recodo.DAL/Entities/Configuration/TeamInvitationConfiguration.cs:11]: You can delete this entity and configuration completely. EF Core 5.0 is smart enough to create a join table it automatically

File: backend/Recodo.DAL/Entities/Configuration/VideoReactionConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Recodo.DAL.Entities.Configuration
+{
+    public class VideoReactionConfiguration : IEntityTypeConfiguration<VideoReaction>
```
Comment (Jyuart) [inline backend/Recodo.DAL/Entities/Configuration/VideoReactionConfiguration.cs]: Also make it owned entity in relation to the video

**Review by Jyuart — CHANGES_REQUESTED** [review]

File: backend/Recodo.DAL/Entities/Configuration/TeamInvitationConfiguration.cs (hunk 6 above)
Comment (VitalikMatvieiev) [reply backend/Recodo.DAL/Entities/Configuration/TeamInvitationConfiguration.cs:11]: But we need more fields than ids in case where EF Core 5.0

Comment (Jyuart) [reply backend/Recodo.DAL/Entities/Configuration/TeamInvitationConfiguration.cs:11]: Aah, yes, you're right!
Then just don't create an id for this entity. We can make it a composite one

File: backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs (hunk 2 above)
Comment (VitalikMatvieiev) [reply backend/Recodo.DAL/Entities/Configuration/CommentReactionConfiguration.cs]: But if I understand clearly, we should have property with Type Comment, and in this case we can use 'Owns'

**Review by Jyuart — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**